From 3bbb12eb07c9c162b0b6ef75ef80072ca5e89455 Mon Sep 17 00:00:00 2001
From: aleksey.v.ignatenko <avignate@lab-dyn-116-248.ins.intel.com>
Date: Mon, 19 Aug 2013 15:09:59 +0700
Subject: Dalvik: FPreg32 and FPreg64 types should support debug print

BZ: 131621

Summary:
OP_REM_FLOAT/DOUBLE implementation http://android.intel.com:8080/123858 didnot consider debug
output for the new types FPreg32 and FPreg64 in dvmCompilerPcgGetDtypeName method that can
lead to a failure.

Category: device-enablement
Domain: AOSP-Dalvik-Compiler-CG
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: I1735f5b5385f7a75addcf21abb928eb2375d0e7f
Orig-MCG-Change-Id: Id93f9f8fde2aeac81ddc80b7c2017a99ffe4800b
Signed-off-by: aleksey.v.ignatenko <avignate@lab-dyn-116-248.ins.intel.com>
Reviewed-on: http://android.intel.com:8080/126044
Reviewed-by: Ignatenko, Aleksey V <aleksey.v.ignatenko@intel.com>
Reviewed-by: lab_aqa <lab_aqa@intel.com>
Reviewed-by: Provodin, Vitaly A <vitaly.a.provodin@intel.com>
Reviewed-by: Petrochenko, Dmitry <dmitry.petrochenko@intel.com>
Tested-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Reviewed-by: Hartley, Timothy D <timothy.d.hartley@intel.com>
Reviewed-by: Beyler, Jean Christophe <jean.christophe.beyler@intel.com>
Reviewed-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/compiler/codegen/x86/pcg/Utility.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/vm/compiler/codegen/x86/pcg/Utility.cpp b/vm/compiler/codegen/x86/pcg/Utility.cpp
index 9034b60..fecc498 100644
--- a/vm/compiler/codegen/x86/pcg/Utility.cpp
+++ b/vm/compiler/codegen/x86/pcg/Utility.cpp
@@ -399,7 +399,7 @@ void dvmCompilerPcgDumpModRegInfo (CompilationUnitPCG *cUnit)
 
 const char* dvmCompilerPcgGetDtypeName (pcgDtype dtype)
 {
-    static const char *names[] = {"NOreg", "INTreg", "LLreg", "VXreg32", "DPVXreg64", "Any", "Any4", "Any8"};
+    static const char *names[] = {"NOreg", "INTreg", "LLreg", "VXreg32", "DPVXreg64", "FPreg32", "FPreg64", "Any", "Any4", "Any8"};
 
     if (dtype >= MaxType)
     {
-- 
1.7.4.1

